pkg/webhook/configvalidator/handler.go

```diff
@@ -0,0 +1,201 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package configvalidator
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime/serializer/versioning"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	admissionwebhook "github.com/gardener/gardener/pkg/admissioncontroller/webhook/admission"
+	gardencore "github.com/gardener/gardener/pkg/apis/core"
+	gardencoreinstall "github.com/gardener/gardener/pkg/apis/core/install"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+)
+
+var (
+	gardenCoreScheme  *runtime.Scheme
+	gardenCoreDecoder runtime.Decoder
+)
+
+func init() {
+	gardenCoreScheme = runtime.NewScheme()
+	gardencoreinstall.Install(gardenCoreScheme)
+	gardenCoreDecoder = versioning.NewCodec(nil, serializer.NewCodecFactory(gardenCoreScheme).UniversalDeserializer(),
+		runtime.UnsafeObjectConvertor(gardenCoreScheme), gardenCoreScheme, gardenCoreScheme, nil,
+		runtime.DisabledGroupVersioner, runtime.InternalGroupVersioner, gardenCoreScheme.Name())
+}
+
+// Handler validates configuration part of ConfigMaps which are referenced in Shoot resources.
+type Handler struct {
+	Logger    logr.Logger
+	APIReader client.Reader
+	Client    client.Reader
+	Decoder   admission.Decoder
+
+	ConfigMapKind               string
```

Comment:
Not too strong of an opinion, but the `Kind` suffix is a bit confusing as it is also used to refer to the actual k8s resource `Kind` field. What about using `Purpose`?
```suggestion
	ConfigMapPurpose               string
```